src/engine.c

```diff
@@ -337,10 +337,15 @@ void engine_run(engine_init_flags *init_flags) {
             // Tick dynamic features. This is a dynamically changing tick, and it depends on things such as
             // hit-pause, hit slowdown and game-speed slider. It is meant for ticking everything that has to do
             // with the actual gameplay stuff.
-            has_dynamic = dynamic_wait > game_state_ms_per_dyntick(gs);
+            int dyntick_ms = game_state_ms_per_dyntick(gs);
+            if(debugger_proceed > 0) {
+                dynamic_wait += dyntick_ms;
```

Comment:
I'm not sure it's wise to not increment static_wait here as well? This seems to avoid some things like controller_tick from happening, which breaks REC rewind if you only advance the REC using the number keys now.